documentation/docs/07-misc/07-v5-migration-guide.md

```diff
@@ -699,9 +714,9 @@ In Svelte 4, doing the following triggered reactivity:
 
 This is because the Svelte compiler treated the assignment to `foo.value` as an instruction to update anything that referenced `foo`. In Svelte 5, reactivity is determined at runtime rather than compile time, so you should define `value` as a reactive `$state` field on the `Foo` class. Wrapping `new Foo()` with `$state(...)` will have no effect — only vanilla objects and arrays are made deeply reactive.
 
-### `<svelte:component>` is no longer necessary
+### `<svelte:component>` is no longer necessary {#svelte-component-no-longer-necessary}
```

Comment:
what is the `{#svelte-component-no-longer-necessary}` for?

Reply:
It's a [heading ID](https://www.markdownguide.org/extended-syntax/#heading-ids) so that it can be referenced in  the section  ['Component typing changes'](https://github.com/sveltejs/svelte/blob/f02b415b0893d9756521670a0b97829af695ed2e/documentation/docs/07-misc/07-v5-migration-guide.md#component-typing-changes)  

https://github.com/sveltejs/svelte/blob/f02b415b0893d9756521670a0b97829af695ed2e/documentation/docs/07-misc/07-v5-migration-guide.md?plain=1#L615

I added it since the example block of code used the old dynamic component syntax (and I don't believe this should be changed), so while sequentially reading the migration docs, a bit of a reminder that there's new syntax, where to find it and an example of old vs new is nice.

Reply:
All headings are automatically made linkable. You don't need to do anything extra
```suggestion
### `<svelte:component>` is no longer necessary
```

Reply:
Fair, I customized it since the autogenerated one depends on its parent section as well, and thus seems quite fragile: `#Breaking-changes-in-runes-mode-svelte:component-is-no-longer-necessary`. For example, it will break if this section is moved out of the parent heading.

However, now that I realize, it shouldn't be in the `Breaking-changes-in-runes-mode` section since it's not a breaking change, it just throws a deprecation warning [REPL](https://svelte.dev/playground/hello-world?version=5.1.3#H4sIAAAAAAAAA3VQy2rDMBD8lbUoKIEQk6vrGHrrqT8Q5-DYayKQV0Ja54Hxv0eqAq1De9RoXjuToGZAUYhP1NrA1TjdwQo7xditxUb0SqMXxWESfLeRF4GAP1Uf1m79BTVH7NR4_AtvDTESBxtR-tYpy1VNNavBGsfwNQ4ndNA7M4Dc5un5FMvIy3PojWsR3EjoI6KRIQbBHt48N4wr-V1crt9rKvOfDCrPuypdNkXBvIG7GaVDoJRappyiNaEMhZLAZ-X3U2oxV8HshRCg4FlTuIvxxqJgNwbff_ZZXLOc6PXr10q7LMuW_sf5AS0OXuKpAQAA).

Modifying this PR now to move the section outside Breaking changes